EIPS/eip-3076.md

```diff
@@ -210,72 +197,60 @@ target_epoch <=
+This specification is not backwards-compatible with previous draft versions that used version numbers less than 5.
 
 
-## Implementation
-
-Implementations exist for the following clients:
-- [Teku](https://github.com/ConsenSys/teku/pull/2820)
-- [Lighthouse](https://github.com/sigp/lighthouse/pull/1544)
-- [Prysm](https://github.com/prysmaticlabs/prysm/pull/7518)
-- [Nimbus](https://github.com/status-im/nimbus-eth2/pull/1643)
-
 ## Security Considerations
 
 In order to minimise risk and complexity, the format has been designed to map cleanly onto the internal database formats used by implementers. Nevertheless, there are a few pitfalls worth illuminating.
 
-### Advice for Complete-Record Databases
+### Advice for Complete Databases
 
 For implementers who use a complete record of signed messages to implement their slashing protection database, we make the following recommendations:
 
 * You MUST ensure that, in addition to importing all of the messages from an interchange, all the [conditions](#conditions) are enforced. In particular, conditions (2), (4) and (5) may not have been enforced by your implementation before adopting the interchange format. Our recommendation is to enforce these rules at all times, to keep the implementation clean and minimise the attack surface. For example: your slashing protection mechanism should not sign a block with a slot number less than, or equal to, the minimum slot number of a previously signed block, _irrespective_ of whether that minimum-slot block was imported from an interchange file, or inserted as part of your database's regular operation.
-* If your database records the signing roots of messages in addition to their slot/epochs, you should ensure that imported messages without signing roots are assigned a suitable dummy signing root internally. We suggest using a value like `0x0` which is extremely unlikely to collide with any real signing root.
+* If your database records the signing roots of messages in addition to their slot/epochs, you should ensure that imported messages without signing roots are assigned a suitable dummy signing root internally. We suggest using a special "null" value which is distinct from all other signing roots, although a value like `0x0` may be used instead (as it is extremely unlikely to collide with any real signing root).
+* Care must be taken to avoid signing messages within a gap in the database (an area of unknown signing activity). This could occur if two interchanges were imported with a large gap between the last entry of the first and the first entry of the second. Signing in this gap is not safe, and would violate conditions (2), (4) and (5). It can be avoided by storing an explicit low watermark in addition to the actual messages of the slashing protection database, or by pruning on import so that the oldest messages from the interchange become the oldest messages in the database.
```

Comment:
I also added this security recommendation based on my own experience with Lighthouse. I had implemented slashing protection oblivious to this case until it was pointed out to me by @ajsutton (thanks again :heart:)